### DISCUSSION TITLE
How to write pytest tests for pytorch lightning custom callbacks?

### DISCUSSION THREAD

**Discussion by malfonsoarquimea** [Lightning-AI/pytorch-lightning#13116] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi! I am currently trying to write a set of pytest tests for our codebase, and I have some questions.
My current approach to testing that the training is working as expected was to launch a training process within the pytest test using `subprocess.run()`  with a given set of arguments (in the form of a config file) and then checking that the training process produces the expected files (checkpoints mostly) at the expected directories.
I think that this approach is not appropriate, so I would like to know If any of you could suggest a better way of testing the custom callbacks that I wrote that run when training

Thanks in advance!

**Comment by semaphore-egg** [comment]
Can we imitate the [pl tests](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/callbacks/test_callbacks.py) or something?
